build/build.csproj

```diff
@@ -0,0 +1,19 @@
+<Project Sdk="Microsoft.NET.Sdk">
+
+  <PropertyGroup>
+      <OutputType>Exe</OutputType>
+      <AnalysisMode>Recommended</AnalysisMode>
+      <WarningsAsErrors>true</WarningsAsErrors>
+      <TreatWarningsAsErrors>true</TreatWarningsAsErrors>
+      <CodeAnalysisTreatWarningsAsErrors>true</CodeAnalysisTreatWarningsAsErrors>
+      <EnforceCodeStyleInBuild>true</EnforceCodeStyleInBuild>
+      <Nullable>enable</Nullable>
```

Comment:
I already have this enabled on a separate branch and dealing with the warnings. Let's keep this and something like WarningAsErrors off. Keep the changes simple and progressively move towards better code.

feature/net7.0

Reply:
These extra params in this csproj keep this project locked down with more settings until they can be graduated to the Directory.Build.props for all projects.  

The idea would be to put these in each csproj progressively and when they all match move to Directory.Build.props.  How many steps depends on how big a single PR for fixing warnings, etc should be :)

Reply:
Just for reference, the PR is up (but not building on GA) for nullability and .net 7. As you can see, it's quite large.
https://github.com/Kareadita/Kavita/pull/1677